internal/fsutil/create.go

```diff
@@ -69,7 +77,8 @@ func Create(options *CreateOptions) (*Entry, error) {
 		return nil, err
 	}
 	mode := s.Mode()
-	if o.OverrideMode && mode != o.Mode && o.Mode&fs.ModeSymlink == 0 {
+	if o.OverrideMode && mode != o.Mode && o.Link == "" {
```

Comment:
After discussing it offline, there seems to be a hiccup in two cases:

1. A hardlink points to a symlink. In that case, the mode bits would differ because `o.Mode` will not have the `SymLinkFlag` but `mode` will. 
2. In theory, if we had the following files in the tarball, which ideally we should not face:
```
/file (regular file) with mode 0644
/hardlink (hard link to /file) with mode 0755
```
There comes a decision question about which mode should prevail, 0644 or 0755. 

In that regard, I think it's safe to disable overriding modes for links in general (both hardlink and symlink).

Reply:
We should add a comment here, something like:

```go
// Overrides mode if the entry is not a link and the
// mode differs.
```

Reply:
Thanks, patched.